group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroupMember.java

```diff
@@ -0,0 +1,756 @@
+            this.rebalanceTimeoutMs = rebalanceTimeoutMs.orElse(this.rebalanceTimeoutMs);
+            return this;
+        }
+
+        public Builder setClientId(String clientId) {
+            this.clientId = clientId;
+            return this;
+        }
+
+        public Builder setClientHost(String clientHost) {
+            this.clientHost = clientHost;
+            return this;
+        }
+
+        public Builder setState(MemberState state) {
+            this.state = state;
+            return this;
+        }
+
+        public Builder setTopologyEpoch(int topologyEpoch) {
+            this.topologyEpoch = topologyEpoch;
+            return this;
+        }
+
+        public Builder maybeUpdateTopologyEpoch(OptionalInt topologyEpoch) {
+            this.topologyEpoch = topologyEpoch.orElse(this.topologyEpoch);
+            return this;
+        }
+
+        public Builder setProcessId(String processId) {
+            this.processId = processId;
+            return this;
+        }
+
+        public Builder maybeUpdateProcessId(Optional<String> processId) {
+            this.processId = processId.orElse(this.processId);
+            return this;
+        }
+
+        public Builder setUserEndpoint(StreamsGroupMemberMetadataValue.Endpoint userEndpoint) {
+            this.userEndpoint = userEndpoint;
+            return this;
+        }
+
+        public Builder maybeUpdateUserEndpoint(Optional<StreamsGroupMemberMetadataValue.Endpoint> userEndpoint) {
+            this.userEndpoint = userEndpoint.orElse(this.userEndpoint);
+            return this;
+        }
+
+        public Builder setClientTags(Map<String, String> clientTags) {
+            this.clientTags = clientTags;
+            return this;
+        }
+
+        public Builder maybeUpdateClientTags(Optional<Map<String, String>> clientTags) {
+            this.clientTags = clientTags.orElse(this.clientTags);
+            return this;
+        }
+
+        public Builder setAssignment(Assignment assignment) {
+            this.assignedActiveTasks = assignment.activeTasks();
+            this.assignedStandbyTasks = assignment.standbyTasks();
+            this.assignedWarmupTasks = assignment.warmupTasks();
+            return this;
+        }
+
+        public Builder setAssignedActiveTasks(Map<String, Set<Integer>> assignedActiveTasks) {
+            this.assignedActiveTasks = assignedActiveTasks;
+            return this;
+        }
+
+        public Builder setAssignedStandbyTasks(Map<String, Set<Integer>> assignedStandbyTasks) {
+            this.assignedStandbyTasks = assignedStandbyTasks;
+            return this;
+        }
+
+        public Builder setAssignedWarmupTasks(Map<String, Set<Integer>> assignedWarmupTasks) {
+            this.assignedWarmupTasks = assignedWarmupTasks;
+            return this;
+        }
+
+        public Builder setAssignmentPendingRevocation(Assignment assignment) {
```

Comment:
This is a bit of a weird function. Why is a whole assignment pending revocation? I'm not totally against it and may have actually added it, but we could check if we really need it.

Reply:
We use it here: https://github.com/apache/kafka/blob/c64fa8d5cd2d020c25d89a38211f380dbaf4306d/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java#L3996